src/modules/AnalysisExecution/components/SubmissionCodeViewer/container.js

```diff
@@ -43,16 +43,20 @@ export class SubmissionCode extends FileLoader {
     if (this.props.urlParams.fileId) {
       super.componentWillMount();
     }
-    this.props.loadBreadcrumbs({
-      entityType: this.props.from,
-      id: this.props.submissionGroupId || this.props.submissionId,
-    });
   }
 
   componentWillReceiveProps(nextProps) {
     if (this.props.params.fileId !== nextProps.params.fileId) {
       this.loadData(nextProps);
     }
+    if (!this.props.submission.id && this.props.submission.id !== nextProps.submission.id) {
```

Comment:
`if  (!A && A!== B)` ?
This conditinal looks fishy. Can you demonstrate both branches in action?

Reply:
My mistake. The idea is to get breadcrumbs only after getting submission id.
Fixed to !A && B